sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -176,6 +153,40 @@ object FileFormatWriter extends Logging {
     //    V1 write command will be empty).
     if (Utils.isTesting) outputOrderingMatched = orderingMatched
 
+    SQLExecution.checkSQLExecutionId(sparkSession)
+
+    val finalStatsTrackers = if (writeFilesOpt.isDefined) {
+      val writeFilesMetrics = writeFilesOpt.get.metrics
+      val finalMetrics = statsTrackers match {
+        case Seq(tracker: BasicWriteJobStatsTracker) =>
```

Comment:
should we use `statsTrackers.find`? in case there are more than one trackers in `statsTrackers` (seems possible w.r.t. the code)

Reply:
It does not possible since all v1 write pass one tracker and we only do that if WriteFiles exists. But I'm fine to change it to be compatible with developers.